src/installer/url/installer.go

```diff
@@ -0,0 +1,73 @@
+package url
+
+import (
+	"fmt"
+
+	"github.com/Dynatrace/dynatrace-operator/src/dtclient"
+	"github.com/Dynatrace/dynatrace-operator/src/dtclient/types"
+	"github.com/Dynatrace/dynatrace-operator/src/installer/symlink"
+	"github.com/Dynatrace/dynatrace-operator/src/processmoduleconfig"
+	"github.com/spf13/afero"
+)
+
+type Properties struct {
+	Os           string
+	Arch         string
+	Type         string
+	Flavor       string
+	Version      string
+	Technologies []string
+	Url          string // if this is set all settings before it will be ignored
+}
+
+func (props *Properties) fillEmptyWithDefaults() {
+	if props.Technologies == nil || len(props.Technologies) == 0 {
+		props.Technologies = []string{"all"}
+	}
+}
+
+func NewUrlInstaller(fs afero.Fs, dtc dtclient.Client, props *Properties) *urlInstaller {
+	return &urlInstaller{
+		fs:    fs,
+		dtc:   dtc,
+		props: props,
+	}
+}
+
+type urlInstaller struct {
+	fs    afero.Fs
+	dtc   dtclient.Client
+	props *Properties
+}
```

Comment:
Struct should be above constructor.

Also:
```suggestion
type urlInstaller struct {
	filesystem    afero.Fs
	dynatraceClient   dtclient.Client
	properties *Properties
}
```